### DISCUSSION TITLE
Enable an optional string identifier property so callback can properly identify a step

### DISCUSSION THREAD

**Discussion by justin-hackin** [gilbarbara/react-joyride#672] [Ideas]
I have a tour that visits a particular element more than once. I need to perform an action upon reaching this element the first time, thus I can't use the target property of the step for identification. Elsewhere it has been suggested to use the title property for identification but I don't want a title on my step. I could use the index of the step but then the application would break if I added a step at the top of the list. Thus, I'm proposing this library enable an optional string property `id` in the typescript definitions for Step. Do you think there a better solution to the problem though? If not, and this proposal is approved, I could put in a quick PR

**Comment by gilbarbara** [comment]
hey @justin-hackin 

You could filter your steps by the target selector and match the tour index with the first occurrence of the filtered steps.

**Reply by justin-hackin** [reply]
That still suffers from the brittleness I spoke of with index targeting, though less.  I just extended the Step definition myself and it works fine.
